### ISSUE TITLE
random word masking index shouble be great than 0

### ISSUE THREAD

**Issue by Dicer-Zz** [huggingface/transformers#15342]
## Information

https://github.com/huggingface/transformers/blob/05fa1a7ac17bb7aa07b9e0c1e138ecb31a28bbfe/src/transformers/data/data_collator.py#L771

This line should be:
```python
random_words = torch.randint(1, len(self.tokenizer), labels.shape, dtype=torch.long)
```
because random word masking index should be great than 0 (0 means [PAD] in bert's vocabulary).

I found the bug when I want to change masked input_ids to tokens using tokenizer.convert_ids_to_tokens. If replace a masked id by 0, and then convert_ids_to_tokens, all tokens after 0 will be discarded.

## Environment info

- `transformers` version: 4.15.0
- Platform: Linux-5.4.0-62-generic-x86_64-with-debian-buster-sid
- Python version: 3.7.2
- PyTorch version (GPU?): 1.10.1 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <False>
- Using distributed or parallel set-up in script?: <False>

**Comment by LysandreJik** [comment]
cc @sgugger

**Comment by sgugger** [comment]
I don't think there is anything in the research article or the original pretraining code that excludes the pad token from the random words, so I would leave this as is. You can use your own data collator with any modification you want :-)
